### PR TITLE
Make python language server exit when parent process dies

### PR DESCRIPTION
**Pull request by jiahuijiang**

### CODE REVIEW

File: pyls/_utils.py [hunk 1]
```diff
@@ -113,3 +113,19 @@ def clip_column(column, lines, line_number):
     # https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#position
     max_column = len(lines[line_number].rstrip('\r\n')) if len(lines) > line_number else 0
     return min(column, max_column)
+
+
+def is_process_alive(pid):
+    """ Check whether the process with the given pid is still alive.
+
+    :param pid: (int): process id
```
Comment (gatesn) [inline pyls/_utils.py]: Can you use Google-style docstrings instead?
```
Args:
    pid (int): process ID
```
```

File: pyls/_utils.py [hunk 2]
```diff
@@ -113,3 +113,19 @@ def clip_column(column, lines, line_number):
...
+    """
+    try:
+        os.kill(pid, 0)
+    except OSError as err:
+        # no such process or process is already dead
+        log.info("process %s is not alive: %s", pid, err)
```
Comment (gatesn) [inline pyls/_utils.py]: Bit agressive to log info this?

File: pyls/python_ls.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Copyright 2017 Palantir Technologies, Inc.
 import logging
 import socketserver
+from threading import Thread, Timer
```
Comment (gatesn) [inline pyls/python_ls.py]: Just `import threading`

File: pyls/_utils.py (hunk 2 above)
Comment (jiahuijiang) [reply pyls/_utils.py]: 👍 moving the logging to ```watch_parent_process```

**Comment by jiahuijiang** [comment]
@gatesn done :D

**Review by gatesn — APPROVED** [review]

**Pull request merged**